pkg/virt-launcher/virtwrap/network/network.go

```diff
@@ -61,7 +61,7 @@ type PodCacheInterface struct {
 // downgrade privileges for virt-launcher, specifically, to remove NET_ADMIN
 // capability. Future patches should address that. See:
 // https://github.com/kubevirt/kubevirt/issues/3085
-type NetworkInterface interface {
+type PlugPhases interface {
```

Comment:
I think an interface should be a noun in its single form.
Making it plural hints it contains something.

If this interface represents something (I find it hard to see something useful really), then it can be `NetworkConfig` or `PodNetworkConfig`, whatever that means.

Reply:
I find it very hard to fathom what this interface represents. I tried to make it very clear in the commit message
```
    An instance of network.NetworkInterface does not represent an actual
    network interface; it is just an aggregation of two methods, used to
    plug a network interface to a pod. However in a few cases, the name
    `vif` is unjustly attached to such instance.
```
I don't think that it is a NetworkConfig. It is just a pare of loosely-related functions exposed so they easier to use by gomock. I is hard to find a non-awkward name to an entity with such an awkward raison d'etre. How about `PlugPhasePair`?

Reply:
I was thinking that one can saw: I pass you a NetworkConfig, on which you can plug (execute) two phases to apply it.
Maybe `NetworkConfigurer` ? The configurer has two plug-gable phases it can execute to apply it.

Reply:
It seems to me that you are trying to force logic into where it does not exist. I spent hours trying to understand how this `NetworkInterface` relates to similarly-named entities in the project. I failed. I don't want to mislead future readers to assume that there exists an entity named `NetworkConfig` that can be passed from a producer to a consumer and has any kind of standing. This `PlugPhases` is an aggregate of two functions, with no meaning beyond the possibility to mock them.

I hope that in a future PR, someone smarter than me can rewrite the tests so they do not require this clog.

Reply:
Ok, I give up.

Reply:
> I think an interface should be a noun in its single form.

I agree with Eddy.

@dankenigsberg @EdDev what do you think about -
1. change the name of `NetworkInterface` to `PodNic`.
2. change the name of `PodInterface` to `PodNicImpl`
3. Move the parameters passed to `PlugPhase1\2` to be data members of `PodInterfaceNicImpl`. So that `PodInterfaceNicImpl` won't be just an aggregation of two methods but actually will have a state.
4. In a follow-up PR/commit `{Bridge,Masquerade,Slirp}PodInterface` will contain `PodInterface` as a data member.
5. In a follow-up PR I would also change `PlugPhase1\2` to -
`PlugPhase1`-> `ConfigurePodInterfaceBindingViaVirtHandler`
`PlugPhase2`-> `ConfigurePodInterfaceBindingViaVirtLauncher`

Reply:
I think you are suggestion now: `PodNic.PlugPhase1`.
The problem is that it does not represent really a nic, the action is on all nics/interfaces of the VMI. (that is why I tried to suggest something generic like network-config.

At the end, I would like to see the phases just gone.
I do not think anyone should group the actions on the handler or on the launcher... the "grouping" is just the fact that they are called in the correct place/flow.

I think all suggestions are good if they move us one bit forward, so I would agree with anything at this point :)

Reply:
`PodNic.PlugPhase1` is on a specific nic. It is not on all nics/interfaces.

Anyway, after re-thinking, point 4 is problematic as `PodNic.PlugPhase1` is creating the binding so it cannot pass itself as a parameter to the binding. So ignore point 4. The other are still relevant.

Reply:
Thank you for clarifying and correcting me here. Passing in the VMI confused me.
As such, I like this option. The name `PodNIC` could also be `PodInterface` but it is a nit and both will work out for me.

Reply:
The reason I chose `Nic` and not `Interface` is to avoid confusion with the golang interface.

Reply:
> 3. Move the parameters passed to PlugPhase1\2 to be data members of PodInterfaceNicImpl. So that PodInterfaceNicImpl won't be just an aggregation of two methods but actually will have a state.

To recap, do you suggest
```
PodNICImpl struct {
 vmi *v1.VirtualMachineInstance
 iface *v1.Interface
 network *v1.Network
 podInterfaceName string
}
``` 
(`pid` does not belong there, I assume) ?

The factory would have to be changed to fill these (somewhat redundant) fields. I feel that it is going to be a very long crawl in a deep rat hole until an object worthy of the `PodNIC` name can be created, which I'm not yet sure if should have one at all. I certainly cannot do this in this rename-only PR.

The ugly name `PlugPhases` is meant as a stopgap, a mark saying "this is not an object. this is bad. do not assume anything about this interface because nothing is there". Maybe `PodNIC` is obscure enough not to cause assumptions. I'll try.

Reply:
joining Phase1/2 renames suggested by @AlonaKaplan and @maiqueb for future consideration

> PlugPhase1-> ConfigurePodInterfaceBindingViaVirtHandler SetupPodNetworkTrustedComponentPhase or SetupPodNetworkPrivilegedPhase
> PlugPhase2-> ConfigurePodInterfaceBindingViaVirtLauncher SetupPodNetworkUntrustedComponentPhase or SetupPodNetworkUnprivilegedPhase